MicrobitGPIO.py

```diff
@@ -26,3 +27,29 @@ def write_analog(self, pin, val):
             print("Invalid Pin number: should be 0,1,2: ", err)
         except ValueError as err:
             print("Invalid Analog value: should be in range 0 to 1023: ", err)
+    
+    def write_digital(self, pin, val):
+        pin = int(pin)
+        val = int(val)
+        try:
+            if val not in range(DIGITAL_VAL_MAX):
+                raise ValueError
+            cmd = self.pin[pin] + ".write_digital(" + str(val) +")\r\n"
+            self.ser.write(cmd.encode("utf-8"))
+        except IndexError as err:
+            print("Invalid Pin number: should be 0,1,2: ", err)
+        except ValueError as err:
+            print("Invalid Digital value: should be either 0 or 1: ", err)
+
+    def set_analog_period(self, pin, period):
+        pin = int(pin)
+        period = int(period)
+        try:
+            if period < 150 or period > 25000:
```

Comment:
Magic numbers?
Also why are we restricting the period to this range?

Reply:
It Should be in the range of only 150us to 25ms. PWM output in analog pin between 150us to 25ms